lib/reporters/tap.js

```diff
@@ -77,3 +79,95 @@ function TAP(runner) {
 function title(test) {
   return test.fullTitle().replace(/#/g, '');
 }
+
+/**
+ * Inherit from `Base.prototype`.
+ */
+inherits(TAP, Base);
+
+/**
+ * Initialize a new `TAPProducer`. Should only be used as a base class.
+ *
+ * @private
+ * @class
+ * @api private
+ */
+function TAPProducer() {}
+
+TAPProducer.prototype.writePending = function(n, test) {
+  console.log('ok %d %s # SKIP -', n, title(test));
+};
+
+TAPProducer.prototype.writePass = function(n, test) {
+  console.log('ok %d %s', n, title(test));
+};
+
+TAPProducer.prototype.writeEnd = function(runner) {
+  console.log('# tests ' + (runner.stats.passes + runner.stats.failures));
```

Comment:
Just noticed this, but think this is probably an unreported existing bug.
`runner.stats.tests` will not agree with this math if pending tests occurred.